tests/python_functional/src/syslog_ng_config/syslog_ng_config.py

```diff
@@ -112,3 +117,11 @@ def create_example_msg_generator(self, **options):
         generator_source.DEFAULT_MESSAGE = "-- Generated message. --"
         generator_source.options = options
         return generator_source
+
+    def create_app_parser(self, topic, **options):
+        opts = options.copy()
```

Comment:
It works without it, but the user would not expect that this function will modifiy its input parameter. Performance-wise it does not really matter, but copying results in safer code. I would stick with this version.

Reply:
Sorry, I would add another comment here. I understand that why the `topic` option is a separate parameter here (because you wanted to help test writers), but I think it is not necessary. I would say if the test writer did not know the mandatory option of the related config object (driver/parser), he/she should write a minimal testcase with that driver (which means to call a `config.create_my_config_object()` method with empty parameters from the testcase) and syslog-ng will describe which mandatory option is missing (at this point run a half written test is not takes much time, testcase log should contains why syslog-ng can not started). Maybe this is not nice, but the config API (in long-term) can be more clean. If we follow your way we need to be consequent in all drivers/parsers/config objects (which is a lot). If you remember I started to eliminate this kind of config parameters from config methods in my previous PR (which I closed).
I understand that you wanted to help the test writers (imho: +1), but where will be the borderline?
I wanted to keep the API as flexible as can and it can be achieve only with simplifications.
At here you added a hint: test writer must define a topic option, but I can not writer such test where I want to define app-parser without any topic (and the topic="" is not the same as creating an app-parser without a topic option like: app-parser(); ).
And if we can eliminate the topic parameter from the method the copy will also did not needed anymore.

Reply:
Fixed.